lang/cs/Org.Apache.REEF.Network.Examples/Elastic/WithFailures/ElasticBroadcastDriverWithFailures.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Network.Elastic.Driver;
+using Org.Apache.REEF.Common.Tasks;
+using Org.Apache.REEF.Network.Elastic.Topology.Logical.Enum;
+using Org.Apache.REEF.Network.Elastic.Driver.Default;
+using Org.Apache.REEF.Network.Elastic.Failures;
+using Org.Apache.REEF.Network.Elastic.Failures.Default;
+using Org.Apache.REEF.Network.Elastic.Task.Default;
+
+namespace Org.Apache.REEF.Network.Examples.Elastic
+{
+    /// <summary>
+    /// Example implementation of broadcasting using the elastic group communication service.
+    /// </summary>
+    public sealed class ElasticBroadcastDriverWithFailures<TSlave>
+        : DefaultElasticDriver
+        where TSlave : DefaultElasticTask
+    {
+        [Inject]
+        private ElasticBroadcastDriverWithFailures(
+            string stageName,
+            int numEvaluators,
+            IElasticContext context) : base(context)
+        {
+            IFailureStateMachine failureMachine = new DefaultFailureStateMachine();
+
+            failureMachine.SetThreasholds(new Tuple<IFailureState, float>[]
+            {
+                Tuple.Create(
+                    new DefaultFailureState(
+                        (int)DefaultFailureStates.ContinueAndReconfigure) as IFailureState, 0.01F),
+                Tuple.Create(
+                    new DefaultFailureState(
+                        (int)DefaultFailureStates.ContinueAndReschedule) as IFailureState, 0.40F),
+                Tuple.Create(
+                    new DefaultFailureState(
+                        (int)DefaultFailureStates.StopAndReschedule) as IFailureState, 0.60F),
+                Tuple.Create(
+                    new DefaultFailureState(
+                        (int)DefaultFailureStates.Fail) as IFailureState, 0.80F)
+            });
```

Comment:
Alternatively, you can make `SetThreshold()` return `this` and chain up the methods:
```C#
failureMachine
    .SetThreshold(new DefaultFailureState(DefaultFailureStates.ContinueAndReconfigure), 0.01F)
    .SetThreshold(new DefaultFailureState(DefaultFailureStates.ContinueAndReschedule), 0.40F)
    .SetThreshold(new DefaultFailureState(DefaultFailureStates.StopAndReschedule), 0.60F)
    .SetThreshold(new DefaultFailureState(DefaultFailureStates.Fail), 0.80F);
```

Reply:
I implemented the first option. Chains of `SetThreshold` can be problematic because the change is checked for consistency (the levels must be ordered) at each modification. With `SetThreshold` I only check the final consistency and is more appropriate in this situation. Let me also add a comment in method's doc about this.